### ISSUE TITLE
"No suitable spell-checker program found"

### ISSUE THREAD

**Issue by bersbersbers** [r-lib/devtools#2376]
**Is your feature request related to a problem? Please describe.**
Running `devtools::check(incoming = T, quiet = T)` simply returns `0 errors ✔ | 0 warnings ✔ | 0 notes ✔` although the package I am checking has numerous problems.

The reason can (only) be seen with `devtools::check(incoming = T, quiet = F)`:
```
...
─  checking CRAN incoming feasibility ...Error in aspell(files, filter = list("dcf", ignore = ignore), control = control,  : 
     No suitable spell-checker program found
   Execution halted
   
── R CMD check results ──...
Duration: 1.3s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔
> 
```

**Describe the solution you'd like**
Indicate that checks have not been completed, maybe by adding an error.

**Comment by bersbersbers** [comment]
Just adding a few bits of information here:

- `R -q -e "utils::aspell('run.R')"` works fine, and so does `... "utils::aspell(NULL)"`
- `aspell` is not installed globally as far as I can see:

```
$ aspell
If 'aspell' is not a typo you can use command-not-found to lookup the package that contains it, like this:
    cnf aspell
```

**Comment by bersbersbers** [comment]
The second problem is this:

`R CMD check` uses `utils::aspell(program = "aspell")`:
https://github.com/wch/r-source/blob/b6b0a445f44b51507281ef48e032b18606c25ebc/src/library/tools/R/QC.R#L7102-L7108

But `devtools` checks for `utils::aspell(program = NULL)`:
https://github.com/r-lib/devtools/blob/d5901f8b1b986cfc623a3615e511015ad468ab0f/R/check.R#L241-L246

In my case, 
```
> utils:::aspell_find_program(NULL)
           hunspell 
"/usr/bin/hunspell" 
> utils:::aspell_find_program("aspell")
<NA> 
  NA 
```

So `utils::aspell("run.R", program = NULL)` works, but `utils::aspell("run.R", program = "aspell")` does not.

So in summary, there is two issues:

1. `devtools` should (also?) check `!is.na(utils:::aspell_find_program("aspell"))`
2. If `_R_CHECK_CRAN_INCOMING_USE_ASPELL_` is set (for whatever reason - even manually by the user) `R CMD check` fails due to `aspell` failing, this should be reported as an error.

**Comment by hadley** [comment]
PR fixes 1; rcmdcheck issue tracks 2.
